packages/transition-frontend/src/services/definitions/DefinitionsService.ts

```diff
@@ -5,25 +5,55 @@
  * License text available at https://opensource.org/licenses/MIT
  */
 
-import React from 'react';
 import serviceLocator from 'chaire-lib-common/lib/utils/ServiceLocator';
 import * as Status from 'chaire-lib-common/lib/utils/Status';
+import { de } from 'date-fns/locale';
 import { Dictionary } from 'lodash';
 
+// Cache the definitions once they are fetched. This is to avoid fetching the
+// same definition multiple times. 'loading' means a request to the server is
+// ongoing and further requests for the same label can wait for it to return.
+// 'error' means the server returned an error for this definition. If the
+// definition is not in this cache, it has not been fetched yet.
+const definitionCache: { [label: string]: Dictionary<string> | 'loading' | 'error' } = {};
+
 export const getDefinitionFromServer = (
     label: string,
-    setDefinition: React.Dispatch<React.SetStateAction<Dictionary<string>>>,
-    setGotError: React.Dispatch<React.SetStateAction<boolean>>
+    setDefinition: (arg: Dictionary<string>) => void,
+    setGotError: (arg: boolean) => void
 ) => {
-    serviceLocator.socketEventManager.emit(
-        'service.getOneDefinition',
-        label,
-        (definitionFromServer: Status.Status<Dictionary<any>>) => {
-            if (Status.isStatusOk(definitionFromServer)) {
-                setDefinition(Status.unwrap(definitionFromServer));
-            } else {
-                setGotError(true);
+    if (definitionCache[label] === 'loading') {
+        // Wait for the previous call to complete
+        const interval = setInterval(() => {
+            if (definitionCache[label] !== 'loading') {
+                clearInterval(interval);
+                if (definitionCache[label] === 'error') {
```

Comment:
We won't do it now, but maybe something to think about later. 
In case of error, we should note the timestamp and retry at a moment later.